src/Avatar/Avatar.js

```diff
@@ -22,6 +22,7 @@ function getStyles(props, context) {
       borderRadius: '50%',
       height: size,
       width: size,
+      boxShadow: src ? `inset 0px 0px 0px 1px ${avatar.borderColor}` : 'none',
```

Comment:
I have tried this path in the past. But we ended up reverting this change for rendering performances reasons https://github.com/callemall/material-ui/pull/1204#issuecomment-123796638.
I'm not sure it worth it.